orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/SavePipelineTask.java

```diff
@@ -92,9 +92,17 @@ public TaskResult execute(Stage stage) {
     outputs.put("application", pipeline.get("application"));
     outputs.put("pipeline.name", pipeline.get("name"));
 
-    if (pipeline.containsKey("id")) {
-      // a newly created pipeline will not yet have an `id`
-      outputs.put("pipeline.id", pipeline.get("id"));
+    try {
+      Map<String, Object> savedPipeline = (Map<String, Object>) objectMapper.readValue(
+        response.getBody().in(), Map.class
+      );
+      outputs.put("pipeline.id", savedPipeline.get("id"));
+    } catch (Exception e) {
+      log.error("Unable to deserialize saved pipeline, reason: ", e.getMessage());
```

Comment:
really just here to handle a situation where front50 hasn't been deployed and we need to fallback to the previous behavior of trying to fetch it out of the existing pipeline.